extensions/wikia/Wall/WallHooksHelper.class.php

```diff
@@ -2327,4 +2295,81 @@ public static function onNamespaceIsMovable( bool &$result, int $ns ): bool {
 		return true;
 	}
 
+	/**
+	 * SUS-1554: Prevent invalid content manipulation in Wall Namespaces
+	 *
+	 * @param Title $title
+	 * @param User $user
+	 * @param string $action
+	 * @param array|null $result error message array (message name and optional params) or null
+	 * @return bool true if action is not handled or allowed, false if action should be prevented
+	 */
+	public static function onGetUserPermissionsErrors( Title $title, User $user, string $action, &$result ): bool {
+		$ns = $title->getNamespace();
+		if ( $ns === NS_USER_WALL_MESSAGE_GREETING ) {
+			return static::checkWallGreeting( $title, $user, $action, $result );
+		}
+
+		if ( !WallHelper::isWallNamespace( $ns ) ) {
+			return true;
+		}
+
+		global $wgIsValidWallTransaction, $wgCommandLineMode;
+		$isValidContext = $wgIsValidWallTransaction || $wgCommandLineMode;
+		$allow = true;
+
+		if ( !$isValidContext ) {
+			switch ( $action ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a "case default" clause to this "switch" statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS131)